plugins/single_plugins/idle.cpp

```diff
@@ -361,15 +350,38 @@ class wayfire_idle_singleton : public wf::singleton_plugin_t<wayfire_idle>
         return true;
     };
 
+    wf::signal_connection_t fullscreen_state_changed{[this] (wf::signal_data_t *data)
+    {
+        bool state = data ? true : false;
```

Comment:
This will not work with multiple outputs and multiple fullscreen views, right? You can have fullscreen on both outputs, then if one of them is unfullscreened ...

Maybe you can change idle_enabled to an int counter to keep track of this.

Reply:
I made a counter for fullscreen but kept the toggle function independent so it would always work.